static/dispatch.ml

```diff
@@ -15,25 +15,32 @@ module Dispatch (C: CONSOLE) (FS: KV_RO) (S: HTTP) = struct
 
   let read_fs fs name =
     FS.size fs name >>= function
-    | `Error (FS.Unknown_key _) ->
-      Lwt.fail (Failure ("read " ^ name))
-    | `Ok size ->
-      FS.read fs name 0 (Int64.to_int size) >>= function
-      | `Error (FS.Unknown_key _) -> Lwt.fail (Failure ("read " ^ name))
-      | `Ok bufs -> Lwt.return (Cstruct.copyv bufs)
+      | `Error (FS.Unknown_key _) ->
+        Lwt.fail (Failure ("read " ^ name))
+      | `Ok size ->
+        FS.read fs name 0 (Int64.to_int size) >>= function
+        | `Error (FS.Unknown_key _) -> Lwt.fail (Failure ("read " ^ name))
+        | `Ok bufs -> Lwt.return ((Magic_mime.lookup name), (Cstruct.copyv bufs))
 
   (* dispatch files *)
-  let rec dispatcher fs ?header uri = match Uri.path uri with
-    | "" | "/" -> dispatcher fs ?header (Uri.with_path uri "index.html")
-    | path ->
-      let mimetype = Magic_mime.lookup path in
-      let headers = Cohttp.Header.add_opt header "content-type" mimetype in
-      Lwt.catch
-        (fun () ->
-           read_fs fs path >>= fun body ->
-           S.respond_string ~status:`OK ~body ~headers ())
-        (fun exn ->
-           S.respond_not_found ())
+  let dispatcher fs ?header uri = 
+    let path = 
+      let p = Uri.path uri in 
+      match p.[ (String.length p) - 1 ] with  
```

Comment:
you should check that the path is not empty (e.g. if `String.length p = 0 then p else ...`)